pkg/provider/bao/bao.go

```diff
@@ -129,6 +130,12 @@ func (p *Provider) LoadSecrets(ctx context.Context, paths []string) ([]provider.
 
 	secretInjector := injector.NewSecretInjector(p.injectorConfig, p.client, p.secretRenewer, slog.Default())
 	inject := func(key, value string) {
+		// Check deduplication
+		if utils.IsKeyDuplicated(&sanitized.secrets, key) {
+			slog.Warn(fmt.Sprintf("Deduplication detected for key: %s, overriding it", key))
+			utils.RemoveSecretByKey(&sanitized.secrets, key)
```

Comment:
if we construct the sanitised slice, then we dont need to remove the key if it is already present. Instead, we skip adding it in case its already present. WDYT?

Reply:
That would be the better approach, but checking this out one more time:
- First the environ is checked for secrets: `secretInjector.InjectSecretsFromVault(vaultEnviron, inject)`
- Second the `*_FROM_PATH`: `secretInjector.InjectSecretsFromVaultPath(p.fromPath, inject)`

So if we keep the current order of injection, we need to override, but if we change the order and check `
*_FROM_PATH` first and then the environ, we can use this approach.

Reply:
We can keep it as is for now